### PR TITLE
Rdcc_1873_1:get the email from header instead of query param for email masking

### PR DESCRIPTION
**Pull request by KOTLAPRA23**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-1873

### Change description ###

feat:get the email from header instead of query param for email masking

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/userprofileapi/controller/UserProfileController.java [hunk 1]
```diff
@@ -180,7 +184,8 @@
     @ApiOperation(value = "Retrieves a User Profile and their Roles with the given Email Address",
             authorizations = {
                     @Authorization(value = "ServiceAuthorization"),
-                    @Authorization(value = "Authorization")
+                    @Authorization(value = "Authorization"),
+                    @Authorization(value = "User-Email")
             }
```
[the remark this answers is not in the document]
Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/userprofileapi/controller/UserProfileController.java:189]: ok. ideally all header like content-type and many more are in two words.

[the remark this answers is not in the document]
Comment (shashank-rastogi) [reply src/main/java/uk/gov/hmcts/reform/userprofileapi/controller/UserProfileController.java:189]: okay.. then lets make it all in small letters like -> user-email
I was suggesting to follow same pattern which we use for any custom headers like ServiceAuthorization

[the remark this answers is not in the document]
Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/userprofileapi/controller/UserProfileController.java:189]: ok.done

[the remark this answers is not in the document]
Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/userprofileapi/controller/UserProfileController.java:189]: to test from swagger we need to add header .

**Review by mallikarjunputtana — APPROVED** [review]

**Review by pksaxena01 — APPROVED** [review]
Approved

**Pull request merged**